modules/rostests/apitests/shell32/GUIDFromString.cpp

```diff
@@ -0,0 +1,82 @@
+/*
+ * PROJECT:     ReactOS api tests
+ * LICENSE:     LGPL-2.1-or-later (https://spdx.org/licenses/LGPL-2.1-or-later)
+ * PURPOSE:     Tests for GUIDFromStringA/W
+ * COPYRIGHT:   Copyright 2024 Katayama Hirofumi MZ <katayama.hirofumi.mz@gmail.com>
+ */
+
+#include <shelltest.h>
+#include <initguid.h>
+#include <undocshell.h>
+
+DEFINE_GUID(invalid_guid, 0xDEADDEAD, 0xDEAD, 0xDEAD, 0xED, 0xED, 0xED, 0xED,
+            0xED, 0xED, 0xED, 0xED);
+
+//DEFINE_GUID(IID_IShellLinkW, 0x000214F9, 0x0000, 0x0000, 0xC0, 0x00, 0x00, 0x00,
+//            0x00, 0x00, 0x00, 0x46);
+//DEFINE_GUID(IID_IShellLinkW_Invalid, 0x000214F9, 0x0000, 0x0000, 0xC0, 0x00, 0x00, 0x00,
+//            0x00, 0x00, 0x00, 0xED);
+
+static void TEST_GUIDFromStringA(void)
+{
+    GUID guid;
+
+    guid = invalid_guid;
+    ok_int(GUIDFromStringA("", &guid), FALSE);
```

Comment:
should it crash when receiving a `nullptr` for either of the arguments?

Reply:
Fixed.